### PR TITLE
Add check updates workflow

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
- Add check updates workflow for grafana-dashboard
- Imported existing grafana-dashboard from [sunbeam project](https://opendev.org/openstack/sunbeam-charms/src/branch/main/charms/openstack-exporter-k8s/src/grafana_dashboards/openstack-exporter.json)
- Add Makefile targets and scripts

Depends on #16.

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout local repository
+        uses: actions/checkout@v4
+      - name: Checkout remote repository
+        working-directory: ../
+        run: |
+          git clone https://opendev.org/openstack/sunbeam-charms
+      - name: Check differences
+        run: |
```
Comment (chanchiwai-ray) [inline .github/workflows/pull-request.yaml:51]: Welcome suggestions for the script :)

File: src/grafana_dashboards/openstack-exporter.json
Comment (chanchiwai-ray) [inline src/grafana_dashboards/openstack-exporter.json:1]: Need to import it here, otherwise CI will fail for sure

**Review by Pjack — APPROVED** [review]
LGTM

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (jneo8) [reply .github/workflows/pull-request.yaml:51]: +1 for moving the script to script file.

**Review by jneo8 — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -36,3 +36,24 @@ jobs:
...
+          LOCAL_SOURCE=./src/grafana_dashboards/openstack-exporter.json
+          REMOTE_SOURCE=../sunbeam-charms/charms/openstack-exporter-k8s/src/grafana_dashboards/openstack-exporter.json
+          diff $LOCAL_SOURCE $REMOTE_SOURCE > /dev/null 2>&1
+          DIFF_STATUS=$?
+          if [ $DIFF_STATUS -ne 0 ]; then
+            echo "Differences detected in the file ./src/grafana_dashboards/openstack-exporter.json"
+            exit 1
+          fi
```
[the remark this answers is not in the document]
Comment (chanchiwai-ray) [reply .github/workflows/pull-request.yaml:59]: Sure, though `make check-dashboards-sync` seems not really necessary, as we can do `./scripts/check-dashboards-sync`, @jneo8 do you have a preference?

**Review by jneo8 — APPROVED** [review]

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (chanchiwai-ray) [reply .github/workflows/pull-request.yaml]: Done with Makefile + scripts

File: .github/workflows/pull-request.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chanchiwai-ray) [reply .github/workflows/pull-request.yaml]: Done, and documented

**Review by Pjack — APPROVED** [review]

**Comment by chanchiwai-ray** [comment]
@samuelallan72 we are closing as completed because the functionality requirement is mostly (if not all) achieved, but feel free to open follow-up PRs to address issue you found

**Pull request merged**